tensorboard/webapp/widgets/line_chart_v2/lib/chart_types.ts

```diff
@@ -0,0 +1,61 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+import {
+  DataSeries,
+  DataSeriesMetadataMap,
+  Dimension,
+  Extent,
+} from './internal_types';
+import {RendererType} from './renderer/renderer_types';
+import {ScaleType} from './scale_types';
+
+export interface Chart {
+  resize(dim: Dimension): void;
+
+  setMetadata(metadataMap: DataSeriesMetadataMap): void;
+
+  setViewBox(extent: Extent): void;
+
+  setData(data: DataSeries[]): void;
+
+  setXScaleType(type: ScaleType): void;
+
+  setYScaleType(type: ScaleType): void;
+
+  dispose(): void;
+}
+
+export interface ChartCallbacks {
+  onDrawEnd(): void;
+}
+
+export interface BaseChartOption {
+  callbacks: ChartCallbacks;
+  domDimension: Dimension;
+}
+
+export interface SvgChartOption extends BaseChartOption {
+  type: RendererType.SVG;
+  container: SVGElement;
+}
+
+export interface WebGlChartOption extends BaseChartOption {
+  type: RendererType.WEBGL;
+  devicePixelRatio: number;
+  container: OffscreenCanvas | HTMLCanvasElement;
+}
+
+export type ChartOption = SvgChartOption | WebGlChartOption;
```

Comment:
Should be `Options`, not `Option`—likewise for other types and variable
names. (This is a collection of options, not a single option.)